src/web/app/components/dialogs/CodeGenerator/QRCodeGenerator.spec.tsx

```diff
@@ -19,46 +19,41 @@ jest.mock('helpers/useI18n', () => () => ({
 }));
 
 const mockOpen = jest.fn();
+
 jest.mock('implementations/browser', () => ({
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   open: (...args: any) => mockOpen(...args),
 }));
 
 const mockInsertImage = jest.fn();
+
 jest.mock('app/actions/beambox/svgeditor-function-wrapper', () => ({
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   insertImage: (...props: any) => mockInsertImage(...props),
 }));
 
-const mockOnClose = jest.fn();
 describe('test QRCodeGenerator', () => {
   it('should behave correctly', () => {
-    const { baseElement } = render(<QRCodeGenerator onClose={mockOnClose} />);
+    const [isInvert, setIsInvert] = [false, jest.fn()];
+    const [text, setText] = ['', jest.fn()];
+    const { baseElement } = render(
+      <QRCodeGenerator
+        isInvert={isInvert}
+        setIsInvert={setIsInvert}
+        text={text}
+        setText={setText}
+      />
+    );
```

Comment:
The input content will not change using constant props.
Remove `fireEvent.change(input, { target: { value: 'some text' } });
    expect(baseElement).toMatchSnapshot();`;
Or wrap `QRCodeGenerator` and states together (example: `src/web/app/views/beambox/Right-Panels/ObjectPanelItem.spec.tsx`).